mail/spring-cloud-starter-stream-source-mail/src/test/java/org/springframework/cloud/stream/app/mail/source/MailSourceConfigurationTests.java

```diff
@@ -0,0 +1,179 @@
+package org.springframework.cloud.stream.app.mail.source;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.util.concurrent.TimeUnit;
+
+import org.hamcrest.Matchers;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.IntegrationTest;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.cloud.stream.app.test.mail.PoorMansMailServer;
+import org.springframework.cloud.stream.app.test.mail.PoorMansMailServer.ImapServer;
+import org.springframework.cloud.stream.app.test.mail.PoorMansMailServer.Pop3Server;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.messaging.Message;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * Tests for Mail Source Configuration.
+ *
+ * @author Amol
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = MailSourceConfigurationTests.MailSourceApplication.class)
+@DirtiesContext
+@WebIntegrationTest(randomPort = true)
+public abstract class MailSourceConfigurationTests {
+
+	@Autowired
+	protected Source source;
+
+	@Autowired
+	protected MessageCollector messageCollector;
+
+	@Autowired
+	protected MailSourceProperties properties;
+
+	
+	@IntegrationTest({ "protocol=imap", "port=50000", "username=user", "password=pw",
+			"host=localhost", "folder=INBOX", "mark-as-read=true", "delete=false",
+			"java-mail-properties=mail.imap.socketFactory.fallback=true,mail.store.protocol=imap,mail.debug=true" })
+	public static class ImapPassTests extends MailSourceConfigurationTests {
+		private final static ImapServer imapServer = PoorMansMailServer.imap(50000);
+		
+		@BeforeClass
+		public static void startImapServer() throws Throwable {
+			int n = 0;
+			while (n++ < 100 && (!imapServer.isListening())) {
+				Thread.sleep(100);
+			}
+			assertTrue(n < 100);
+		}
+
+		@Test
+		public void testSimpleTest() throws Exception {
+
+			Message<?> received = messageCollector.forChannel(source.output()).poll(100,
+					TimeUnit.SECONDS);
```

Comment:
M-m-m. 100 seconds seem for me so much to block the bad case on the CI server. 
10 would be enough for our mock mail server.